src/controllers/dlgprefcontrollers.cpp

```diff
@@ -123,49 +132,44 @@ void DlgPrefControllers::setupControllerWidgets() {
                 m_pDlgPreferences,
                 &DlgPreferences::show);
 
-        m_controllerWindows.append(controllerDlg);
-        m_pDlgPreferences->addPageWidget(controllerDlg);
+        m_controllerPages.append(controllerDlg);
 
         connect(pController,
                 &Controller::openChanged,
                 [this, controllerDlg](bool bOpen) {
                     slotHighlightDevice(controllerDlg, bOpen);
                 });
 
-        QTreeWidgetItem * controllerWindowLink = new QTreeWidgetItem(QTreeWidgetItem::Type);
-        controllerWindowLink->setIcon(0, QIcon(":/images/preferences/ic_preferences_controllers.png"));
-        QString curDeviceName = pController->getName();
-        controllerWindowLink->setText(0, curDeviceName);
-        controllerWindowLink->setTextAlignment(0, Qt::AlignLeft | Qt::AlignVCenter);
-        controllerWindowLink->setFlags(Qt::ItemIsSelectable | Qt::ItemIsEnabled);
-        m_pControllerTreeItem->addChild(controllerWindowLink);
-        m_controllerTreeItems.append(controllerWindowLink);
-
-        // Set the font correctly
-        QFont temp = controllerWindowLink->font(0);
+        QTreeWidgetItem* controllerTreeItem = new QTreeWidgetItem(
+                QTreeWidgetItem::Type);
+        m_pDlgPreferences->addPageWidget(controllerDlg,
+                controllerTreeItem,
+                pController->getName(),
+                "controllers.svg");
+
+        m_pControllersRootItem->addChild(controllerTreeItem);
+        m_controllerTreeItems.append(controllerTreeItem);
+
+        // If controller is open make controller label bold
+        QFont temp = controllerTreeItem->font(0);
         temp.setBold(pController->isOpen());
-        controllerWindowLink->setFont(0, temp);
+        controllerTreeItem->setFont(0, temp);
     }
-
-    // If no controllers are available, show the "No controllers available"
-    // message.
-    txtNoControllersAvailable->setVisible(controllerList.empty());
 }
 
-void DlgPrefControllers::slotHighlightDevice(DlgPrefController* dialog, bool enabled) {
-    int dialogIndex = m_controllerWindows.indexOf(dialog);
-    if (dialogIndex < 0) {
+void DlgPrefControllers::slotHighlightDevice(DlgPrefController* controllerDlg, bool enabled) {
+    int controllerPageIndex = m_controllerPages.indexOf(controllerDlg);
```

Comment:
```suggestion
void DlgPrefControllers::slotHighlightDevice(DlgPrefController* pControllerDlg, bool enabled) {
    const int controllerPageIndex = m_controllerPages.indexOf(pControllerDlg);
```

and also add the `p` prefix in the code below both for controllerDlg and controllerTreeItem...

Reply:
Done https://github.com/mixxxdj/mixxx/pull/3484/commits/f9742c598320341a2e5d96fe3c972c691213f7d1